### ISSUE TITLE
Startup incompatibility screen does not work with Canary

### ISSUE THREAD

**Issue by DAmNRelentless** [AbdElAziz333/Canary#35]
### Version Information
- canary-mc1.18.2-0.0.9
- forge-40.1.84
- windows 11 pro
- curseforge launcher

### Expected Behavior

When booting the game, a message on a dirt screen should pop up about which mod has compatibility issues according to the respective `mod.toml` file.

This is how it should look:
![](https://i.imgur.com/5DBiGdr.png)

### Actual Behavior

The game won't start if there is a compatibility issue with mods. The window doesn't even open. As an example, the latest version of AE2 (appliedenergistics2-forge-11.6.2) with an older version of JEI (jei-1.18.2-9.7.2.281) can be used. The AE2 version relies on v10 of JEI and is not compatible with v9.7 anymore. Usually, this is displayed on startup but with Canary, the game won't start.

The only log message you get that could *slightly* show you what's wrong is:
```
[main/ERROR]: Unsupported installed optional dependencies: jei
```

### Reproduction Steps

> 1. Use two mods that are incompatible with another
> 2. Start the game
> 3. The window won't open

### Other Information

[startup log](https://pastebin.com/hpZMUe59) (a crash report is not generated since the game doesn't even start)

**Comment by AbdElAziz333** [comment]
This issue is not from Canary, the issues you have is other mod compatibilities :

- Better Biome Blend crashes with Rubidium.
- MineTogeather mod crashes Game Menu Mod Option.

**Comment by DAmNRelentless** [comment]
I found out about Canary being the issue by binary search. All other mods are working fine together. The game does not start if Canary is enabled. The same happens with other Lithium forks so I guess this is probably an upstream issue.
The mods you mentioned work pretty fine.

**Comment by DAmNRelentless** [comment]
Just for the sake of proving my point, I created a new instance with just these:
![](https://i.imgur.com/gnCKZ88.png)

And it still happens. It does not happen without Canary. Here's the log from the minimal reproduction setup: https://pastebin.com/a53JE8vY

Please don't shift issues to other mods if you are not a 100% certain about the cause of the issue, please.

**Comment by AbdElAziz333** [comment]
weird, i will look into that.

**Comment by Nuclearsmilz** [comment]
I also experienced a canary crash that I will try to reproduce but was on this same version (canary-mc1.18.2-0.0.9) and same forge (forge-40.1.84). I will try and reproduce the crash but it was related to mixins. Will update this post with crash log.

**Comment by DarkXFast** [comment]
Can also confirm that the mod incompatibility screen doesn't pop up for me too when Canary is installed.

**Comment by AbdElAziz333** [comment]
Sorry for shifting the issue to other mods but i saw the mod errors in the log, i apologize for that!

The issue is fixed in both 0.1.1 for MC 1.19.2 and 0.0.10 for MC 1.18.2.
I hope you have a good day!

**Comment by DAmNRelentless** [comment]
All good, thanks for looking into it.
